stdlib/public/core/DoubleWidth.swift.gyb

```diff
@@ -676,45 +676,7 @@ binaryOperators = [
   }
 }
 
-// FIXME(ABI) (Conditional Conformance):
-// DoubleWidth should conform to SignedInteger where Base : SignedInteger
-extension DoubleWidth where Base : SignedInteger {
-  /// Returns the additive inverse of the specified value.
-  ///
-  /// The negation operator (prefix `-`) returns the additive inverse of its
-  /// argument.
-  ///
-  ///     let x = 21 as DoubleWidth<Int>
-  ///     let y = -x
-  ///     // y == -21
-  ///
-  /// The resulting value must be representable in the same type as the
-  /// argument. In particular, negating a signed, fixed-width integer type's
-  /// minimum results in a value that cannot be represented.
-  ///
-  ///     let z = -DoubleWidth<Int>.min
-  ///     // Overflow error
-  ///
-  /// - Returns: The additive inverse of this value.
-  ///
-  /// - SeeAlso: `negate()`
-  @_inlineable // FIXME(sil-serialize-all)
-  public static prefix func - (_ operand: DoubleWidth) -> DoubleWidth {
-    return 0 - operand
-  }
-  
-  /// Replaces this value with its additive inverse.
-  ///
-  /// The following example uses the `negate()` method to negate the value of
-  /// an integer `x`:
-  ///
-  ///     var x = 21 as DoubleWidth<Int>
-  ///     x.negate()
-  ///     // x == -21
-  ///
-  /// - SeeAlso: The unary minus operator (`-`).
-  @_inlineable // FIXME(sil-serialize-all)
-  public mutating func negate() {
-    self = 0 - self
-  }
-}
+extension DoubleWidth : UnsignedInteger where Base : UnsignedInteger {}
+
+extension DoubleWidth : SignedNumeric, SignedInteger
```

Comment:
Is this a workaround for a compiler issue? Ordinarily, `SignedInteger` implies `SignedNumeric`.

Reply:
No, it's required for conditional conformance. Since there can be multiple paths to implied/inherited protocols, you have to specify them explicitly. (I had to look this up 😄)

Reply:
This seems like a bug. Can you explain why whatever it is that requires this for conditional conformance does not require it for a hypothetical unconstrained `extension CustomDoubleWidth : SignedInteger`? (Let's say for the purposes of this explanation that I wrote a new type `CustomDoubleWidth` where `Base` is always constrained to be a `SignedInteger`.)

Reply:
It's intentional — you can see the rationale in the conditional conformance SE proposal.